### PR TITLE
Add flink-dist LICENSE & NOTICE files

### PR DESCRIPTION
**Pull request by tillrohrmann**
Based on #4.

### CODE REVIEW

File: flink-dist/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+- org.apache.camel:camel-core:2.17.7
+- org.apache.commons:commons-compress:1.4.1
+- org.apache.commons:commons-lang3:3.3.2
+- org.apache.commons:commons-math3:3.5
+- org.javassist:javassist:3.19.0-GA
+- org.objenesis:objenesis:2.1
+- org.rocksdb:rocksdbjni:5.7.5
+- org.xerial.snappy:snappy-java:1.1.4
+
+This project bundles the following dependencies under the BSD license.
```
Comment (StephanEwen) [inline flink-dist/src/main/resources/META-INF/NOTICE:31]: I would combine the BSD licenses in one section (BSD-style licenses). The BSD license names are too varied that I would feel comfortable spelling the specific variant out.

File: flink-dist/src/main/resources/META-INF/NOTICE [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+- org.scala-lang.modules:scala-parser-combinators_2.11:1.0.4
+- org.scala-lang.modules:scala-xml_2.11:1.0.5
+
+This project bundles the following dependencies under the MIT/X11 license.
+See bundled license files for details.
+
+- com.github.scopt:scopt_2.11:3.5.0
+- org.slf4j:slf4j-api:1.7.15
+
+This project bundles the following dependencies under the Public Domain.
```
Comment (StephanEwen) [inline flink-dist/src/main/resources/META-INF/NOTICE]: I would add the full notice text from the project:
```
This project bundles "org.tukaani xz version 1.5".
This Java implementation of XZ has been put into the public domain, thus you can do
whatever you want with it. All the files in the package have been written by Lasse Collin,
but some files are heavily based on public domain code written by Igor Pavlov.
```

Comment (StephanEwen) [reply flink-dist/src/main/resources/META-INF/NOTICE]: That text is the license text from the project homepage.

**Comment by StephanEwen** [comment]
I double checked, all correct.
Only dependency missing is `com.google.code.findbugs:jsr305` which is Apache Licensed, so optional to add.

File: flink-dist/src/main/resources/META-INF/NOTICE (hunk 1 above)
Comment (tillrohrmann) [reply flink-dist/src/main/resources/META-INF/NOTICE:31]: Sounds good. Will change it.

File: flink-dist/src/main/resources/META-INF/NOTICE (hunk 2 above)
Comment (tillrohrmann) [reply flink-dist/src/main/resources/META-INF/NOTICE]: Yes, this is definitely missing. Will add it.

**Comment by tillrohrmann** [comment]
Thanks for the review @StephanEwen. I've addressed your comments. Merging this PR.

**Pull request merged**